File: src/service/ModelBuilder.Core/Models/Project.cs
using System;
using System.Collections.Generic;

namespace Mb.Core.Models
{
    public class Project
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public string Name { get; set; }
        public string Description { get; set; }
        public virtual ICollection<Node> Nodes { get; set; }
        public virtual ICollection<Edge> Edges { get; set; }
    }
}

File: src/client/src/models/classes/Project.ts
import Edge from "./Edge";
import Node from "./Node";

class Project {
    id: string;
    version: string;
    name: string;
    isSubProject: boolean;
    description: string;
    projectOwner: string;
    updatedBy: string;
    update: Date;
    nodes: Node[];
    edges: Edge[];

    // Constructor
    // eslint-disable-next-line @typescript-eslint/no-useless-constructor
    constructor() { }
}

export default Project;
